File: BE/capsule-service/src/main/java/com/timecapsule/capsuleservice/dto/UnopenedCapsuleDto.java
package com.timecapsule.capsuleservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.util.Date;

@Getter
public class UnopenedCapsuleDto {
    private int capsuleId;
    private LocalDate openDate;
    private String address;
    @JsonProperty("isLocked")
    private boolean locked;

    @Builder
    public UnopenedCapsuleDto(int capsuleId, LocalDate openDate, String address, boolean isLocked) {
        this.capsuleId = capsuleId;
        this.openDate = openDate;
        this.address = address;
        this.locked = isLocked;
    }
}

File: FE/AboutTimeCapsule/app/src/main/java/com/aboutcapsule/android/data/capsule/UnopenedCapsuleDto.kt
package com.aboutcapsule.android.data.capsule

import com.google.gson.annotations.SerializedName
import java.time.LocalDate

data class UnopenedCapsuleDto(
    var capsuleId : Int,

    var openDate : String,

    var address : String,

    var isLocked : Boolean

)
